client/app/components/QueryBasedParameterInput.jsx

```diff
@@ -0,0 +1,109 @@
+import { find, isFunction, isNull, isUndefined } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+
+export class QueryBasedParameterInput extends React.Component {
+  static propTypes = {
+    value: PropTypes.any, // eslint-disable-line react/forbid-prop-types
+    queryId: PropTypes.number,
+    onSelect: PropTypes.func,
+  };
+
+  static defaultProps = {
+    value: null,
+    queryId: null,
+    onSelect: () => {},
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      options: [],
+    };
+    this.__loadOptions(this.props.queryId);
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.queryId !== this.props.queryId) {
+      this.__loadOptions(nextProps.queryId, nextProps.value);
+    }
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  __optionsFromQueryResult(queryResult) {
```

Comment:
If it doesn't use this class, why have it as  a class method and not a global function?

Reply:
Well, I decided it's better to access it using `this.__optionsFromQueryResult` instead of `QueryBasedParameterInput.__optionsFromQueryResult`. Maybe, standalone function will be better.

Reply:
You can use `this.constructor. __optionsFromQueryResult` which is a few chars shorter 😛

Reply:
@ranbena Standalone function is even shorter :stuck_out_tongue_winking_eye: